test/gtest/mha_find20.cpp

```diff
@@ -0,0 +1,395 @@
+                         miopenStatusSuccess);
+            EXPECT_EQUAL(miopenGetSolutionSolverId(solution, &solver_id), miopenStatusSuccess);
+        }
+
+        std::cerr << "Finished testing miopenGetSolution<Attribute>." << std::endl;
+    }
+
+    void TestRunSolutionsForward(Handle& handle, const std::vector<miopenSolution_t>& solutions)
+    {
+        std::cerr << "Testing a forward solution..." << std::endl;
+
+        const unsigned int numTensors = tensors.size();
+
+        auto arguments = std::make_unique<miopenTensorArgument_t[]>(numTensors);
+
+        std::vector<miopenTensorDescriptor_t> descVector(numTensors);
+
+        int i = 0;
+        for(const auto& it : tensors)
+        {
+            descVector[i] = &it.second->tensorFloat.desc;
+
+            it.second->gpuBuffer = handle.Write(it.second->tensorFloat.data);
+
+            arguments[i].id         = it.first;
+            arguments[i].descriptor = &descVector[i];
+            arguments[i].buffer     = it.second->gpuBuffer.get();
+
+            ++i;
+        }
+
+        std::vector<miopenTensorArgumentId_t> output_ids = {miopenTensorMhaO,
+                                                            miopenTensorMhaAmaxO,
+                                                            miopenTensorMhaAmaxS,
+                                                            miopenTensorMhaM,
+                                                            miopenTensorMhaZInv};
+
+        std::size_t workspace_size = 0;
+        uint64_t solver_id;
+
+        Workspace workspace;
+
+        for(const auto& solution : solutions)
+        {
+            miopenGetSolutionWorkspaceSize(solution, &workspace_size);
+            miopenGetSolutionSolverId(solution, &solver_id);
+
+            workspace.resize(workspace_size);
+
+            std::cerr << "Run a solution." << std::endl;
+            EXPECT_EQUAL(miopenRunSolution(&handle,
+                                           solution,
+                                           numTensors,
+                                           arguments.get(),
+                                           workspace.ptr(),
+                                           workspace.size()),
+                         miopenStatusSuccess);
+
+            std::cerr << "Solution executed." << std::endl;
+
+            TensorStructMap outputTensorResults;
+
+            // reading results from find 2.0 call and preparing structures for non find 2.0 outputs
+            // to compare with
+            for(const auto& id : output_ids)
+            {
+                const TensorStructPtr& tensorStructPtr = tensors[id];
+                tensorStructPtr->tensorFloat.data      = handle.Read<float>(
+                    tensorStructPtr->gpuBuffer, tensorStructPtr->tensorFloat.data.size());
+
+                TensorStructPtr& ptr = outputTensorResults[id];
+
+                ptr              = TensorStructPtr(new TensorStruct());
```

Comment:
Probably?
```suggestion
                ptr              = std::make_shared<TensorStruct>(true);
```
Or `make_unique`.